### PR TITLE
Aeyb/#129 if the user cannot do something dont show the button

### LINKED ISSUE #129: If the user cannot do something don’t show the button
For our permissions, `MANAGE_MEETINGS`, and `MANAGE_ROLES`, we want to hide the create role and create meeting button if the user doesn't have these permissions

- [x] if the user doesn't have ```view_roles``` permission, don't show them the role's page entirely
- [x] if the user doesn't have ```manage_meetings``` permission, don't show them the new meeting button
- [x] if the user doesn't have ```manage_roles``` button, don't show them the plus sign in roles page

### PR DESCRIPTION
**Pull request by ejiang25**
**Closes:** [#129  ]

**What was done:**
Added a condition for rendering certain components that require specific permissions.

**How to test:**
Create roles with certain perms and check components don't render correctly

**Screenshots of UI changes (optional):**

### CODE REVIEW

**Comment by TylerDonghwi** [comment]
image
Sorry but forgot to spot this here, can you hide this save button if a user doesn't have a "Manage roles" perm too?

File: frontend/src/pages/home/Homepage.tsx [hunk 1]
```diff
@@ -5,31 +5,34 @@ import PostMeeting from './components/PostMeeting';
 import { useMeetingContext } from '../../contexts/MeetingContext';
 import NewMeeting from './components/NewMeeting';
 import { useState } from 'react';
+import { useUserContext } from 'src/contexts/UserContext';
```
Comment (jacobatan) [inline frontend/src/pages/home/Homepage.tsx]: Change this to relative imports

File: frontend/src/pages/profiles/components/Settings.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import IonIcon from '@reacticons/ionicons';
 import React from 'react';
 import { Routes, Route, useNavigate } from 'react-router-dom';
+import { useUserContext } from 'src/contexts/UserContext';
```
Comment (jacobatan) [inline frontend/src/pages/profiles/components/Settings.tsx]: Change this to use relative imports

**Review by jacobatan — COMMENTED** [review]
Minor comments, `PermissionsList.tsx` and `RoleList.tsx` also needs to change to use relative imports.

**Comment by TylerDonghwi** [comment]
yeah everything looks good, works perfectly fine, just change the import directories so it is relative path please.

**Review by TylerDonghwi — APPROVED** [review]
lgtm

**Pull request merged**